internal/controller/freight/finder.go

```diff
@@ -0,0 +1,230 @@
+package freight
```

Comment:
Non-blocking nit: I think that if we would bring this into a more object-oriented shape, we could get rid of some of the repeated code around resolving the origin, while also preventing having to expose two different clients to the health related code.

Reply:
I had thoughts of that as well, but am also trying not to hold up v0.8.0 further. I would be happy to revisit it in the future and I expect our plans for v0.9.0 should give us opportunities to do so.